src/thor/triplegbuilder_utils.h

```diff
@@ -256,36 +257,37 @@ struct MultimodalBuilder {
                                            time_info.second_of_week % kSecondsPerDay);
 
         assumed_schedule = false;
-        uint32_t date, day = 0;
+        uint32_t origin_pivot_days, days_from_creation;
         if (!origin.date_time().empty()) {
-          date = DateTime::days_from_pivot_date(DateTime::get_formatted_date(origin.date_time()));
-
-          if (graphtile->header()->date_created() > date) {
+          origin_pivot_days =
+              DateTime::days_from_pivot_date(DateTime::get_formatted_date(origin.date_time()));
+          days_from_creation = origin_pivot_days - graphtile->header()->date_created();
+
+          // if the departure is in the past or too far in the future, we flag the schedule "assumed"
+          if (graphtile->header()->date_created() > origin_pivot_days ||
+              days_from_creation >
+                  graphtile->GetTransitSchedule(transit_departure->schedule_index())->end_day()) {
             // Set assumed schedule if requested
             if (controller(kNodeTransitPlatformInfoAssumedSchedule)) {
               transit_platform_info->set_assumed_schedule(true);
             }
             assumed_schedule = true;
-          } else {
-            day = date - graphtile->header()->date_created();
-            if (day > graphtile->GetTransitSchedule(transit_departure->schedule_index())->end_day()) {
-              // Set assumed schedule if requested
-              if (controller(kNodeTransitPlatformInfoAssumedSchedule)) {
-                transit_platform_info->set_assumed_schedule(true);
-              }
-              assumed_schedule = true;
-            }
           }
         }
 
         // TODO: all of the duration stuff below assumes the transit departure is on the same day as
         // the origin date time. if the trip took more than one day this will not be the case and
         // negative durations can occur
         if (transit_departure) {
-
-          std::string dt = DateTime::get_duration(origin.date_time(),
-                                                  (transit_departure->departure_time() -
-                                                   (time_info.second_of_week % kSecondsPerDay)),
+          // round up the transit times to full minutes because date_time() will always round down
+          auto round_up_mins = [](uint32_t seconds) {
```

Comment:
so previously it always floored and this always ceilings, therefore we'll still be wrong half the time. but really its all because we dont ahve enough resolution to express what we want. maybe we should just use seconds resolution even though its a bit pedantic and then we dont have to worry about how to display something correctly, or we can round and call it good? anyway not necessary to patch now, a todo is good enough